core/src/raw/oio/write/position_write.rs

```diff
@@ -149,12 +155,16 @@ impl<W: PositionWrite> oio::Write for PositionWriter<W> {
         if let Some(buffer) = self.cache.clone() {
             let offset = self.next_offset;
             self.w.write_all_at(offset, buffer).await?;
+            self.w.as_ref().close().await?;
```

Comment:
We should call `close` no matter there are cache or not.

Reply:
fixed.